### PR TITLE
consumer: Accept configuration from clients

### LINKED ISSUE #19: Allow clients to fully control their configuration
We should find a way to accept from clients a complete librdkafka configuration and use that when creating the consumer/producer.

### PR DESCRIPTION
**Pull request by agis**
Closes #19

### CODE REVIEW

**Comment by agis** [comment]
Tests will be added soon.

File: rafka_test.go [hunk 1]
```diff
@@ -63,6 +65,32 @@ func TestConsumerTopicExclusive(t *testing.T) {
...
 
+func TestConsumerConfig(t *testing.T) {
+	c := newClient("foo:bar1")
+	_, err := c.BLPop(1*time.Second, "topics:foo|auto.offset.reset:latest").Result()
+	if err != nil && err != redis.Nil {
+		t.Fatal(err)
+	}
+
+	// configuration only applies in the 1st operation
+	_, err = c.BLPop(1*time.Second, "topics:foo|auto.offset.reset:malformed").Result()
```
Comment (ctrochalakis) [inline rafka_test.go]: We should also test setting non existing options.

**Comment by dtheodor** [comment]
updated: a client config can be passed as JSON to `BLPOP topics:<topic>|<JSON-encoded consumer config> <timeoutMs>`

it overwrites any values set by the server's consumer config. validation happens the moment the kafka consumer is instantiated, we don't need to do any additional validation on our side

File: README.md [hunk 2]
```diff
@@ -139,7 +139,7 @@ Example using Redis:
 ### Consumer
 - `CLIENT SETNAME <group.id>:<name>`: sets the consumer's group & name
 - `CLIENT GETNAME`
-- `BLPOP topics:<topic> <timeoutMs>`: consumes the next message from the given topic
+- `BLPOP topics:<topic>|<JSON-encoded consumer config> <timeoutMs>`: consumes the next message from the given topic
```
Comment (agis) [inline README.md]: Just a thought: doesn't the following makes it easier to parse?

```
BLPOP topic:<topic> <timeoutMS>|<config>
```

...where the `|...` part is optional?

Comment (agis) [reply README.md]: As we discussed in person, let's keep redis semantics:

```
BLPOP topic:<topic>:<config> <timeoutMS>
```

**Comment by dtheodor** [comment]
Addressed comments

**Pull request merged**